### ISSUE TITLE
Error trying to run ApplyProductWithNegationFunction and ReconstructOracleParameters 

### ISSUE THREAD

**Issue by pferreirafabricio** [MicrosoftDocs/quantum-docs#474]
I'm trying to call the quantum operation using the F# console app in the docs: 

**Quantum code**
```qsharp
namespace Quantum {
    
    open Microsoft.Quantum.Canon;
    open Microsoft.Quantum.Intrinsic;
    
    /// Instantiates the oracle and runs the parameter restoration algorithm.
    operation RunAlgorithm(bits : Bool[]) : Bool[] {
        Message("Hello, quantum world!");
        // construct an oracle using the input array
        let oracle = ApplyProductWithNegationFunction(bits, _, _);
        // run the algorithm on this oracle and return the result
        return ReconstructOracleParameters(Length(bits), oracle);
    }
}

```

**F# App**
```fsharp
namespace Host

open System.Diagnostics
open Quantum
open Microsoft.Quantum.Simulation.Simulators
open Microsoft.Quantum.Simulation.Core

module Host = 
    [<EntryPoint>]
    let main _ =
        printfn "Hello, from classical world"

        use simulator = new QuantumSimulator();

        let parity = Seq.reduce(<>)

        let sampleBits = new QArray<bool>([| false; true; false |]);
        printfn "Samples bits: %A" sampleBits

        let algorithm = RunAlgorithm.Run(simulator, sampleBits).Result
        printfn "Bits from Quantum Computation: %A" algorithm

        // Process the results: in this case, verify that:
        // - the length of the return array equals the length of the input array
        Debug.Assert(algorithm.Length = sampleBits.Length, "Return array length differs from the input array length")
        // - the parity of the returned array matches the parity of the input one
        Debug.Assert((algorithm |> parity) = (sampleBits |> parity), "Return array should have the same parity as the input one")

        0
```

But I'm getting this error: 

```
C:\Users\Fabrício\Desktop\Development - Windows\GitHub - Public\QuantumHelloWorld\Oracle\Quantum\Library.qs(11,22): error QS5022: No identifier with the name "ApplyProductWithNegationFunction" exists. [C:\Users\Fabrício\Desktop\Development - Windows\GitHub - Public\QuantumHelloWorld\Oracle\Quantum\Quantum.csproj]
C:\Users\Fabrício\Desktop\Development - Windows\GitHub - Public\QuantumHelloWorld\Oracle\Quantum\Library.qs(13,16): error QS5022: No identifier with the name "ReconstructOracleParameters" exists. [C:\Users\Fabrício\Desktop\Development - Windows\GitHub - Public\QuantumHelloWorld\Oracle\Quantum\Quantum.csproj]
C:\Users\Fabrício\.nuget\packages\microsoft.quantum.sdk\0.22.187631\Sdk\Sdk.targets(149,5): error MSB3073: The command "dotnet "C:\Users\Fabrício\.nuget\packages\microsoft.quantum.sdk\0.22.187631\DefaultItems\../tools/qsc/qsc.dll" build --format MsBuild -v Normal --response-files obj/qsharp/config/qsc.rsp" exited with code -3. [C:\Users\Fabrício\Desktop\Development - Windows\GitHub - Public\QuantumHelloWorld\Oracle\Quantum\Quantum.csproj]
C:\Users\Fabrício\.nuget\packages\microsoft.quantum.sdk\0.22.187631\Sdk\Sdk.targets(154,5): error : Q# compiler failed with exit code: -3. [C:\Users\Fabrício\Desktop\Development - Windows\GitHub - Public\QuantumHelloWorld\Oracle\Quantum\Quantum.csproj]
```

I also tried use VSCode IntelliSense to import the namespace, but it was not found.

![image](https://user-images.githubusercontent.com/42717522/154475781-e0d7d232-7a53-42b8-ab30-94df8a194b23.png)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c5e6d25a-d9bf-089d-f2e7-0b6b11959bcd
* Version Independent ID: c5e6d25a-d9bf-089d-f2e7-0b6b11959bcd
* Content: [Write a Q# and .NET program to run on a local quantum simulator - Azure Quantum ](https://docs.microsoft.com/en-us/azure/quantum/how-to-csharp-qdk-local?tabs=tabid-cmdline%2Ctabid-csharp)
* Content Source: [articles/how-to-csharp-qdk-local.md](https://github.com/MicrosoftDocs/quantum-docs/blob/main/articles/how-to-csharp-qdk-local.md)
* Service: **azure-quantum**
* Sub-service: **qdk**
* GitHub Login: @cjgronlund
* Microsoft Alias: **cgronlun**

**Comment by pferreirafabricio** [comment]
I resolved it with some code from this website [Quantum computing simulation using Q# with C#](https://chanmingman.wordpress.com/2020/06/20/quantum-computing-simulation-using-q-with-c/), but I think that would be useful to change the sample code in the docs.

Final code:
```qsharp
namespace Quantum {
    
    open Microsoft.Quantum.Arrays;
    open Microsoft.Quantum.Measurement;
    open Microsoft.Quantum.Canon;
    open Microsoft.Quantum.Intrinsic;

    operation ApplyProductWithNegationFunction (vector : Bool[], controls : Qubit[], target : Qubit) : Unit is Adj {
        for (bit, control) in Zipped(vector, controls) {
            (ControlledOnInt(bit ? 1 | 0, X))([control], target);
        }
    }

    operation ReconstructOracleParameters(N : Int, oracle : ((Qubit[], Qubit) => Unit)) : Bool[] {
        use (x, y) = (Qubit[N], Qubit()) {

            oracle(x, y);

            if (N % 2 == 1) {
                X(y);
            }   

            let m = MResetZ(y);

            ResetAll(x);

            return m == One
                ? ConstantArray(N, false) w/ 0 <- true
                | ConstantArray(N, false);
        }
    }
    
    /// Instantiates the oracle and runs the parameter restoration algorithm.
    operation RunAlgorithm(bits : Bool[]) : Bool[] {
        Message("Hello, quantum world!");
        // construct an oracle using the input array
        let oracle = ApplyProductWithNegationFunction(bits, _, _);
        // run the algorithm on this oracle and return the result
        return ReconstructOracleParameters(Length(bits), oracle);
    }
}
```

**Comment by SoniaLopezBravo** [comment]
Hi @tcNickolas, could you take a look at this?

**Comment by tcNickolas** [comment]
The problem is that the docs page references the sample at https://github.com/microsoft/Quantum/blob/main/samples/interoperability/dotnet/qsharp/Operations.qs, but it doesn't quote the complete Q# code, just the last operation from it. If someone tries to reproduce the example looking just at the docs, without clicking through the sample folder to find the Q# code, they are going to get this error indeed, since some of the necessary Q# code is missing. 

Given that we include F# and C# code from the interop example in the docs page completely, I'd recommend to include complete Q# code as well. 

Let me know if you'd like my help sending the PR!

**Comment by pferreirafabricio** [comment]
Resolved with #503. Thanks, @tcNickolas and @SoniaLopezBravo for the help!
